mcp/src/client/base-client.ts

```diff
@@ -0,0 +1,11 @@
+import { CourierClient } from "./courier-client.js";
+
+export class BaseClient {
```

Comment:
If I'm reading it right, `CourierClient` contains many implementations of `BaseClient` (ex. `AudienceClient`), and each `BaseClient` is instantiated with a `CourierClient`.

Could this be simplified so the "child" (`AudienceClient`) doesn't need to reach up to the "parent" (`CourierClient`)?